### ISSUE TITLE
Proposal: Fix naming of NavigationView IsBackButtonVisible property

### ISSUE THREAD

**Issue by michael-hawker** [microsoft/microsoft-ui-xaml#4102]
As discussed on [Twitter here](https://twitter.com/samafshari/status/1357038066968453126?s=20) by @samafshari

# Proposal: Rename NavigationView `IsBackButtonVisible` property to align to its type/conventions

## Summary
Currently `NavigationView` has two properties [`IsBackEnabled`](https://docs.microsoft.com/windows/winui/api/microsoft.ui.xaml.controls.navigationview.isbackenabled?view=winui-2.5) and [`IsBackButtonVisible`](https://docs.microsoft.com/windows/winui/api/microsoft.ui.xaml.controls.navigationview.isbackbuttonvisible?view=winui-2.5). They both have the same documentation string and indicate to the developer that they should be `boolean` values by using the `Is` naming guideline.

However, `IsBackButtonVisible` is an `enum` instead which can be set to `Auto`, `Visible`, or `Collapsed`.

If the `IsBackEnabled` there to be able to show the button but have it not be clickable, then that should be made clearer in the docs. This is probably an issue with the doc string on the `IsBackButtonVisible` property having the same text as `IsBackEnabled`. Otherwise, if it's still about just showing if the button is there or not, it should be _removed_.

I propose that `IsBackButtonVisible` should be `BackButtonVisibility` property which is the `enum` value set. (Or `BackVisibility` if the `IsBackEnabled` property does indeed perform a different action and remains or brevity is important.)

## Rationale
To summarize, there seems to be a few problems here with API consistency:
* Both properties seem to perform similar functions making understanding what they do when combined difficult or know which to use when
* `IsBackButtonVisible` indicated it should be a `boolean`, but is not
* One property is using just `Back` and the other is using `BackButton` as the name of the targeted element.

**Comment by chingucoding** [comment]
Maybe this is something that could be fixed with WinUI 3: https://github.com/microsoft/Xaml-Controls-Gallery/pull/284#issuecomment-562695389? 
@MikeHillberg FYI since this is API surface related

**Comment by chingucoding** [comment]
> If the IsBackEnabled there to be able to show the button but have it not be clickable, then that should be made clearer in the docs. This is probably an issue with the doc string on the IsBackButtonVisible property having the same text as IsBackEnabled. Otherwise, if it's still about just showing if the button is there or not, it should be removed.

This is indeed a doc bug, IsBackEnabled indicates whether you can click the button while IsBackButtonVisible indicates whether its visible.

**Comment by michael-hawker** [comment]
Thanks @chingucoding for the history! It's still also weird that one uses 'Back' and the other 'BackButton'... probably easiest to make one breaking change of converting `IsBackButtonVisible` to `BackVisibility`? Though I feel like normally starting from nothing, the property would have been called `BackButtonVisibility` as was originally thought by the sample writer.

**Comment by chingucoding** [comment]
Yeah the naming is also quite unusual in that area, we should align them to be more consistent with existent naming schemes. I think `BackButtonVisibility` would be the best name here.

**Comment by StephenLPeters** [comment]
The documentations should be updated, however the API name can't be changed without a major release version number. This wont meet our winui3 API change bar, so if it happens it will need to wait for Winui4.

**Comment by chingucoding** [comment]
@michael-hawker Created a PR on the [documentation repo](https://github.com/MicrosoftDocs/winui-api) for this now.

**Comment by mdtauk** [comment]
Is it time to start populating a backlog list of WinUI 4 projects and issues?

**Comment by chingucoding** [comment]
> Is it time to start populating a backlog list of WinUI 4 projects and issues?

There already is one issue label for that: https://github.com/microsoft/microsoft-ui-xaml/issues?q=is%3Aopen+is%3Aissue+label%3AIncludeInWinui4
However projects for that might also be helpful.

**Comment by mdtauk** [comment]
> > Is it time to start populating a backlog list of WinUI 4 projects and issues?
> 
> There already is one issue label for that: https://github.com/microsoft/microsoft-ui-xaml/issues?q=is%3Aopen+is%3Aissue+label%3AIncludeInWinui4
> However projects for that might also be helpful.

I was hoping WinUI 3 would be a stable base to build on - but if there are things off limits for change until a v4 or v5 - what is the release frame for those?

I figured 3.0 - 3.9 - 3.22 | so is 4.0 several years away, or does 3.0 only go up to about 3.4 (as in 4 minor updates a year?)

**Comment by Jaiganeshkumaran** [comment]
> The documentations should be updated, however the API name can't be changed without a major release version number. This wont meet our winui3 API change bar, so if it happens it will need to wait for Winui4.

Well, you can add a new property instead of the same type and then tell in docs not to use the old one. Just like how DisplayMode and PaneDisplayMode coexist in NavigationView. Note the best approach but works.
